File: PythonCode/leetcode/easy/AnagramMappings.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

# @Time    : 2018/1/16 19:28
# @Author  : bearyang
# @Site    : 
# @File    : AnagramMappings.py
# @Software: PyCharm
# 760. Find Anagram Mappings
# https://leetcode.com/problems/find-anagram-mappings/description/


class Solution:
    def anagramMappings(self, A, B):
        """
        :type A: List[int]
        :type B: List[int]
        :rtype: List[int]
        """
        lenA = len(A)
        result = []
        for i in range(lenA):
            for j in range(lenA):
                if A[i] == B[j]:
                    result.append(j)
                    break

        return result


test = Solution()
arrA = [12, 28, 46, 32, 50]
arrB = [50, 12, 32, 46, 28]
result = test.anagramMappings(arrA, arrB)
for item in result:
    print(item)

File: JavaCode/src/com/ysx/leetcode/easy/AnagramMappings.java
package com.ysx.leetcode.easy;

/**
 * 760. Find Anagram Mappings
 * https://leetcode.com/problems/find-anagram-mappings/description/
 * <p>
 * eg.
 * A = [12, 28, 46, 32, 50]
 * B = [50, 12, 32, 46, 28]
 * output:
 * [1, 4, 3, 2, 0]
 */
public class AnagramMappings {

    public static void main(String[] args) {
        AnagramMappings test = new AnagramMappings();
        int[] A = new int[]{12, 28, 46, 32, 50};
        int[] B = new int[]{50, 12, 32, 46, 28};
        int[] result = test.anagramMappings(A, B);
        for (int i : result) {
            System.out.println(i);
        }
    }

    public int[] anagramMappings(int[] A, int[] B) {
        int len = A.length;
        int[] result = new int[len];
        for (int i = 0; i < len; i++) {
            for (int j = 0; j < len; j++) {
                if (A[i] == B[j]) {
                    result[i] = j;
                    break;
                }
            }
        }
        return result;
    }
}
